src/py_genome_dna.cpp

```diff
@@ -64,6 +64,9 @@ GKPY_OMETHOD_BEGIN(GenomeDNA, Call)
 	if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|p", arg_names, &interval_arg, &allow_outside_chromosome))
 		 return nullptr;
 
+	if (!PyInterval::check(interval_arg)) {
+		GK_THROW(type, "Expected argument of type Interval");
+	}
 	// Extract DNA for an Interval arg
 	const interval_t& c = ((PyInterval*)interval_arg)->value();
```

Comment:
Might as well use the helper

```suggestion
	// Extract DNA for an Interval arg
	interval_t c = PyAsInterval(interval_arg);
```

Reply:
Is the type change intentional? What if we keep it as `const interval_t&`? (still compiles and runs ok)

Reply:
either is fine (and basically equivalent) to the compiler since it's a read-only l-value